website/site/content/docs/authentication-backends.md

```diff
@@ -74,28 +73,78 @@ can use one of the community-maintained projects below. Feel free to [submit a p
 | --------------- | -------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- |
-| `repo`          | none                                                           | **Required** for `github` backend; ignored by `git-gateway`. Follows the pattern `[org-or-username]/[repo-name]`.                                    |
+| `repo`          | none                                                           | **Required** for `github` and `gitlab` backends; ignored by `git-gateway`. Follows the pattern `[org-or-username]/[repo-name]`.                                    |
 | `accept_roles`  | none                                                           | `git-gateway` only. Limits CMS access to your defined array of user roles. Omitting this field gives access to all registered users.                 |
 | `branch`        | `master`                                                       | The branch where published content is stored. All CMS commits and PRs are made to this branch.                                                       |
-| `api_root`      | `https://api.github.com` (ignored for `git-gateway` backend)   | The API endpoint. Only necessary in certain cases, like with GitHub Enterprise.                                                                      |
+| `api_root`      | `https://api.github.com` (GitHub) or `https://gitlab.com/api/v4` (GitLab); ignored for `git-gateway`   | The API endpoint. Only necessary in certain cases, like with GitHub Enterprise or self-hosted GitLab.                                                                      |
 | `site_domain`   | `location.hostname` (or `cms.netlify.com` when on `localhost`) | Sets the `site_id` query param sent to the API endpoint. Non-Netlify auth setups will often need to set this for local development to work properly. |
 | `base_url`      | `https://api.netlify.com`                                      | OAuth client URL for the `github` backend. **Required** when using an external OAuth server with the `github` backend.                               |
 | `auth_endpoint` | `auth`                                                         | Path to append to `base_url` for authentication requests. Optional.                                                                                  |
+| `auth_url` | `https://gitlab.com/oauth/authorize` on `gitlab` backend; ignored by `github` and `git-gateway`                                                         | Authentication endpoint URL. **Required** for implicit grant on self-hosted GitLab                                                                                  |
```

Comment:
@erquhart @Benaiah @talves Should we split this into `base_url` and `auth_endpoint` to be consistent with the other backends?